src/OpenDiffix.Core/QueryValidator.fs

```diff
@@ -48,14 +48,7 @@ let private validateSubQuery selectQuery =
   validateSelectTarget selectQuery
```

Comment:
Notice how this becomes mutually recursive. The caller controls when (and if) that happens.